### PR TITLE
Command line parameter fixes and cmd config section

### PR DESCRIPTION
**Pull request by goedzo**
This PR uses full screen mode when the yuzu is called with a rom as an argument. This is usable for front-ends like hyperspin / retrofe.

Also a partial fix to read the proper section related to the #1906 issue.

### CODE REVIEW

File: src/video_core/renderer_opengl/gl_shader_decompiler.cpp [hunk 1]
```diff
@@ -2891,10 +2891,8 @@ class GLSLGenerator {
                 const bool depth_compare =
                     instr.texs.UsesMiscMode(Tegra::Shader::TextureMiscMode::DC);
                 const auto process_mode = instr.texs.GetTextureProcessMode();
-
-                UNIMPLEMENTED_IF_MSG(instr.texs.UsesMiscMode(Tegra::Shader::TextureMiscMode::NODEP),
-                                     "NODEP is not implemented");
-
+                /*UNIMPLEMENTED_IF_MSG(instr.texs.UsesMiscMode(Tegra::Shader::TextureMiscMode::NODEP),
+                                     "NODEP is not implemented");*/
```
Comment (DarkLordZach) [inline src/video_core/renderer_opengl/gl_shader_decompiler.cpp]: I'm sorry, what?
Do not comment out asserts like this in a PR

File: src/yuzu/main.cpp [hunk 2]
```diff
@@ -15,6 +15,7 @@
 #include "core/hle/service/acc/profile_manager.h"
 #include "core/hle/service/am/applets/applets.h"
 
+
```
Comment (DarkLordZach) [inline src/yuzu/main.cpp]: why is this necessary

File: src/yuzu/main.cpp [hunk 3]
```diff
@@ -91,6 +92,7 @@ static FileSys::VirtualFile VfsDirectoryCreateFileWrapper(const FileSys::Virtual
 #include "yuzu/main.h"
 #include "yuzu/ui_settings.h"
 
+
```
Comment (DarkLordZach) [inline src/yuzu/main.cpp]: ditto

File: src/yuzu_cmd/config.cpp [hunk 4]
```diff
@@ -226,11 +226,12 @@ static const std::array<int, 8> keyboard_mods{
 void Config::ReadValues() {
     // Controls
     for (std::size_t p = 0; p < Settings::values.players.size(); ++p) {
-        const auto group = fmt::format("ControlsP{}", p);
+        const auto group = fmt::format("Controls", p);
```
Comment (DarkLordZach) [inline src/yuzu_cmd/config.cpp]: no, this is incorrect.
doing this prevents multiple player's input settings from being saved, instead amalgamating them into one.

File: src/yuzu/main.cpp [hunk 5]
```diff
@@ -1805,6 +1813,7 @@ int main(int argc, char* argv[]) {
     QApplication::setAttribute(Qt::AA_DontCheckOpenGLContextThreadAffinity);
     QApplication app(argc, argv);
 
+
```
Comment (DarkLordZach) [inline src/yuzu/main.cpp]: why?

File: src/yuzu/main.cpp [hunk 6]
```diff
@@ -199,6 +201,12 @@ GMainWindow::GMainWindow()
     QStringList args = QApplication::arguments();
     if (args.length() >= 2) {
         BootGame(args[1]);
+        // Enter full screen mode
+        // Don't confirm closing
+        UISettings::values.confirm_before_closing = false;
```
Comment (DarkLordZach) [inline src/yuzu/main.cpp]: this may not be desired behavior, perhaps add a `-f` or similar flag for full screen

File: src/video_core/renderer_opengl/gl_shader_decompiler.cpp [hunk 7]
```diff
@@ -3784,136 +3782,142 @@ class GLSLGenerator {
...
+                        return ProgramResult{generator.GetShaderCode(), generator.GetEntries()};
+                    } catch (const DecompileFail& exception) {
+                        LOG_ERROR(HW_GPU, "Shader decompilation failed: {}", exception.what());
+                    }
+                    return {};
+                }
 
-} // namespace OpenGL::GLShader::Decompiler
\ No newline at end of file
+            } // namespace OpenGL::GLShader::Decompiler
```
Comment (DarkLordZach) [inline src/video_core/renderer_opengl/gl_shader_decompiler.cpp]: why did you just increase the indentation of this section?

File: src/yuzu/main.cpp (hunk 6 above)
Comment (FearlessTobi) [reply src/yuzu/main.cpp]: This could also possibly break QT fullscreen code.

File: src/video_core/renderer_opengl/gl_shader_decompiler.cpp [hunk 8]
```diff
@@ -3784,136 +3782,142 @@ class GLSLGenerator {
      * @return the offset of the next instruction to compile. PROGRAM_END if the program
      * terminates.
      */
-    u32 CompileRange(u32 begin, u32 end) {
```
Comment (ReinUsesLisp) [inline src/video_core/renderer_opengl/gl_shader_decompiler.cpp]: Why are you moving this methods on an unrelated commit (and unrelated PR)?

Comment (goedzo) [reply src/video_core/renderer_opengl/gl_shader_decompiler.cpp]: Looks like visual studio autoformatted this method for me. Was not my intention to change any indent.

File: src/video_core/renderer_opengl/gl_shader_decompiler.cpp (hunk 7 above)
Comment (goedzo) [reply src/video_core/renderer_opengl/gl_shader_decompiler.cpp]: It looks like visual studio auto formatting did this for me. I did not wanted to change any identation here.

File: src/yuzu/main.cpp (hunk 2 above)
Comment (goedzo) [reply src/yuzu/main.cpp]: It's not. Can be deleted.

File: src/yuzu/main.cpp (hunk 3 above)
Comment (goedzo) [reply src/yuzu/main.cpp]: not needed.

File: src/yuzu/main.cpp (hunk 6 above)
Comment (goedzo) [reply src/yuzu/main.cpp]: This is added so that when you run it onto a frontend (like hyperspin, or retrofe), you can have it headless when exitiing the app with alt-f4. Problem now is that the pop-up in full-screen mode doesn't appear, and you cannot confirm the message box.

File: src/yuzu/main.cpp (hunk 5 above)
Comment (goedzo) [reply src/yuzu/main.cpp]: Otherwise you will not automatically go into fullscreen mode, when a rom is added as a parameter.

File: src/yuzu_cmd/config.cpp (hunk 4 above)
Comment (goedzo) [reply src/yuzu_cmd/config.cpp]: I had to re-use the logic from the yuzu gui version, that deals with it in the same way as used in the qt-config.ini file. One or the other is incorrect, so they should match a similar behaviour, If you look at the multiple player's input setting in qt-config.ini you see that this uses now the same logic.

Comment (goedzo) [reply src/yuzu_cmd/config.cpp]: @DarkLordZach could you please guide me to the right way to get this configured? Are we following yuzu-gui in how it's using qt-config.ini? Or do we keep it as a seperated configuration structure? Or do we need to change the gui to follow cmd? I don't know the dev / direction guidelines between cmd and gui, so any feedback would be appreciated.

**Comment by goedzo** [comment]
Ok, I fixed all formatting and clang issues, and merged all commits into one single. This PR should be ok to merge now.

**Comment by goedzo** [comment]
@Hexagon12 and @DarkLordZach , I've merged all your feedback into a new commit. Please validate if now ok. Please note that I cannot help the blank line at the end of "gl_shader_decompiler.cpp", since the Github browser version automatically adds this, even if I delete it :(.

File: src/yuzu_cmd/config.cpp (hunk 4 above)
Comment (DarkLordZach) [reply src/yuzu_cmd/config.cpp]: The current config format works fine for both, for SDL I don't see an issue with 
    `[ControlsP2]`
    `a = <keycode>`
    `b = <keycode>`

If anything, this was done so the SDL config file would be easier to manually edit (as its split by player) versus the Qt config which wasn't really intended to be manually edited (use the UI).

File: src/yuzu/main.cpp (hunk 6 above)
Comment (DarkLordZach) [reply src/yuzu/main.cpp]: This does not check for any sort of flags to determine if full screen is the desired option, additionally make all of the assumptions for confirm before closing hidden behind options. I don't remember if confirm closing is a UI option rn, but if it is, why couldnt a user just set that and then set a new option in UI for boot to fullscreen automatically?
Ultimatiely, what I'd like to avoid is coding for one specific use case (esp. because hyperspin and retrofe are not the only frontends that might use something like this)

**Comment by bunnei** [comment]
@DarkLordZach please re-review, thank you!

**Comment by FearlessTobi** [comment]
To clarify again, this can't be merged until the two pending comments are addressed.

**Comment by Hexagon12** [comment]
@goedzo Are you still planning on to continue working on this PR? If yes, could you please rebase the PR?

**Comment by oblitum** [comment]
Hi, sorry to noise on this PR. Is there a way to run `yuzu-cmd` and have it recognize my gamepad? Dos this PR tries to fix that? I currently need that because not only I prefer to start games from another frontend, but also now when I start from yuzu GUI it's segfaulting on badalloc after allocating more than 10GB, `yuzu-cmd` just allocates 6GB.

**Comment by DarkLordZach** [comment]
Closing as unresponsive
